server/src/main/java/io/crate/expression/scalar/string/TrimFunctions.java

```diff
@@ -284,8 +283,10 @@ public String evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Str
     }
 
     private static String trimChars(String target, String charsToTrimArg, TrimMode mode) {
-        HashSet<Character> charsToTrim =
-            new HashSet<>(Chars.asList(charsToTrimArg.toCharArray()));
+        HashSet<Character> charsToTrim = new HashSet<>();
+        for (char c : charsToTrimArg.toCharArray()) {
+            charsToTrim.add(c);
+        }
```

Comment:
Could consider using `CharArraySet` here

Reply:
`CharArraySet` does not implement Set<Character>, so this would have a few implications on calling functions. But sure, lets do it.